presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -207,7 +228,26 @@ public synchronized String toString()
                 .toString();
     }
 
-    private synchronized void addRunnerThread()
+    private synchronized void addRunnerThreadsIfNeeded(int targetRunnerThreadsCount)
+    {
+        targetRunnerThreads = targetRunnerThreadsCount;
+
+        for (int i = 0; i < targetRunnerThreadsCount - runnerThreads; i++) {
+            addRunnerThreadInternal();
+            runnerThreads++;
+        }
```

Comment:
If addRunnerThreadInternal is not expected to fail in normal operation, then just set runnerThreads = targetRunnerThreadsCount after the loop is done.